### ISSUE TITLE
Unable to compile for @electron target

### ISSUE THREAD

**Issue by vladimiry** [kriszyp/msgpackr-extract#3]
Use command to trigger compilation (see https://www.electronjs.org/docs/tutorial/using-native-node-modules): `HOME=~/.electron-gyp node-gyp rebuild --target=13.1.4 --arch=x64 --dist-url=https://electronjs.org/headers`

Output:
```
gyp info it worked if it ends with ok
gyp info using node-gyp@8.1.0
gyp info using node@15.14.0 | linux | x64
gyp info find Python using Python version 3.9.5 found at "/usr/bin/python3"
gyp http GET https://electronjs.org/headers/v13.1.4/node-v13.1.4-headers.tar.gz
gyp http 200 https://gh-contractor-zcbenz.s3.amazonaws.com/atom-shell/dist/v13.1.4/node-v13.1.4-headers.tar.gz
gyp http GET https://electronjs.org/headers/v13.1.4/SHASUMS256.txt
gyp http 200 https://gh-contractor-zcbenz.s3.amazonaws.com/atom-shell/dist/v13.1.4/SHASUMS256.txt
gyp info spawn /usr/bin/python3
gyp info spawn args [
gyp info spawn args   '/usr/lib/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/home/usr/dev/_temp/msgpackr-extract/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/usr/lib/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/home/usr/.electron-gyp/.cache/node-gyp/13.1.4/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/home/usr/.electron-gyp/.cache/node-gyp/13.1.4',
gyp info spawn args   '-Dnode_gyp_dir=/usr/lib/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/home/usr/.electron-gyp/.cache/node-gyp/13.1.4/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/home/usr/dev/_temp/msgpackr-extract',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make: Entering directory '/home/usr/dev/_temp/msgpackr-extract/build'
  CXX(target) Release/obj.target/msgpackr-extract/src/extract.o
In file included from /home/usr/.electron-gyp/.cache/node-gyp/13.1.4/include/node/v8.h:30,
                 from ../src/extract.cpp:8:
/home/usr/.electron-gyp/.cache/node-gyp/13.1.4/include/node/v8-internal.h: In function ‘void v8::internal::PerformCastCheck(T*)’:
/home/usr/.electron-gyp/.cache/node-gyp/13.1.4/include/node/v8-internal.h:454:38: error: ‘remove_cv_t’ is not a member of ‘std’; did you mean ‘remove_cv’?
  454 |             !std::is_same<Data, std::remove_cv_t<T>>::value>::Perform(data);
      |                                      ^~~~~~~~~~~
      |                                      remove_cv
/home/usr/.electron-gyp/.cache/node-gyp/13.1.4/include/node/v8-internal.h:454:38: error: ‘remove_cv_t’ is not a member of ‘std’; did you mean ‘remove_cv’?
  454 |             !std::is_same<Data, std::remove_cv_t<T>>::value>::Perform(data);
      |                                      ^~~~~~~~~~~
      |                                      remove_cv
/home/usr/.electron-gyp/.cache/node-gyp/13.1.4/include/node/v8-internal.h:454:50: error: template argument 2 is invalid
  454 |             !std::is_same<Data, std::remove_cv_t<T>>::value>::Perform(data);
      |                                                  ^
/home/usr/.electron-gyp/.cache/node-gyp/13.1.4/include/node/v8-internal.h:454:63: error: ‘::Perform’ has not been declared
  454 |             !std::is_same<Data, std::remove_cv_t<T>>::value>::Perform(data);
      |                                                               ^~~~~~~
In file included from ../src/extract.cpp:9:
../src/extract.cpp: At global scope:
/home/usr/.electron-gyp/.cache/node-gyp/13.1.4/include/node/node.h:790:7: warning: cast between incompatible function types from ‘void (*)(v8::Local<v8::Object>)’ to ‘node::addon_context_register_func’ {aka ‘void (*)(v8::Local<v8::Object>, v8::Local<v8::Value>, v8::Local<v8::Context>, void*)’} [-Wcast-function-type]
  790 |       (node::addon_context_register_func) (regfunc),                  \
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/usr/.electron-gyp/.cache/node-gyp/13.1.4/include/node/node.h:808:3: note: in expansion of macro ‘NODE_MODULE_CONTEXT_AWARE_X’
  808 |   NODE_MODULE_CONTEXT_AWARE_X(modname, regfunc, NULL, 0)
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/extract.cpp:269:1: note: in expansion of macro ‘NODE_MODULE_CONTEXT_AWARE’
  269 | NODE_MODULE_CONTEXT_AWARE(extractor, initializeModule);
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
make: *** [msgpackr-extract.target.mk:133: Release/obj.target/msgpackr-extract/src/extract.o] Error 1
make: Leaving directory '/home/usr/dev/_temp/msgpackr-extract/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (node:events:369:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:290:12)
gyp ERR! System Linux 5.12.9-1-MANJARO
gyp ERR! command "/home/usr/.nvm/versions/node/v15.14.0/bin/node" "/usr/bin/node-gyp" "rebuild" "--target=13.1.4" "--arch=x64" "--dist-url=https://electronjs.org/headers"
gyp ERR! cwd /home/usr/dev/_temp/msgpackr-extract
gyp ERR! node -v v15.14.0
gyp ERR! node-gyp -v v8.1.0
gyp ERR! not ok 
```

I've noticed some native modules I use go with https://github.com/nodejs/node-addon-api rather than `nan` which if I got it right is considered as a forward-looking approach.

**Comment by kriszyp** [comment]
Ok, I attempted to make a fix (to make the build configured to use c++14 for newer electron), can you try with v1.0.12?

**Comment by vladimiry** [comment]
I've just tested the master branch, there is no difference, same error output (test command listed in the original message).

**Comment by kriszyp** [comment]
If possible, would you mind sending me your `msgpackr-extract/build/config.gypi` file?

**Comment by vladimiry** [comment]
Here it's [config.gypi.txt](https://github.com/kriszyp/msgpackr-extract/files/6750495/config.gypi.txt).

Also tried to recompile (using the same `HOME=~/.electron-gyp node-gyp rebuild --target=13.1.4 --arch=x64 --dist-url=https://electronjs.org/headers` command) https://github.com/atom/node-keytar (uses https://github.com/nodejs/node-addon-api but at the moment downgrading `node-addon-api` to 3.1.0 is required, the respective issue has been placed there) and https://github.com/vladimiry-playground/node-spellchecker (uses `nan`) and the outcome was good.

**Comment by kriszyp** [comment]
Hmm, I tried to remove the C++ version altogether, does that help at all?

**Comment by vladimiry** [comment]
I've pulled d4be333. Have not tested it in runtime yet but it's compilable now. So the issue seems to be resolved.

**Comment by vladimiry** [comment]
Btw I tested the compiling on linux system only so far.

**Comment by kriszyp** [comment]
ok, well, maybe I have been over-complicated things with the C++ version specification.

**Comment by vladimiry** [comment]
Another simple way to test compiling is to run `npx prebuildify --napi --strip`.

**Comment by vladimiry** [comment]
I've pinned db33cbc for now (code line is here https://github.com/vladimiry/ElectronMail/blob/b2566eff12bf734d6e1a83a32e92f38c487bbd77/package.json#L268) which appears to be compilable on all platforms. So please feel free to close the issue if you think there is nothing to tweak in its scope.

**Comment by kriszyp** [comment]
Ok, I published this in v1.0.13, along with an update to msgpackr too.
